src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -266,7 +266,7 @@ public boolean checkSolidity(Hash hash, boolean milestone, int maxProcessedTrans
 
                         if (!transactionRequester.isTransactionRequested(hashPointer, milestone)) {
                             transactionRequester.requestTransaction(hashPointer, milestone);
-                            break;
+                            continue;
```

Comment:
not sure if the following suggestion makes sense:
if it's a milestone, then `continue`, else `break`.

the reason being:
this mechanism can be used to fill the request queue with arbitrary data faster.
```
0
| \
0  ?
0
| \
0  ?
0
| \
0  ?
...
```
in the current code the first `?` would be requested and stop (and if this `?` does not exist the rest will never be requested, with this change all the historical `?`s will be requested (up until the `maxProccessedTransactions` is reached).

I'm not sure this is such an issue, but my proposal would not change behavior for random transactions but achieve the improvement for syncing nodes.

Reply:
I think Alon's point is that all branches that are not confirmed by a milestone are "arbitrary data" and we don't want to have priority time or resources on them. The requester queue can fill up with sidetangle requests and thus hinder the solidification of "milestone branches".

However, if the `TipsSolidifier` is not enabled we are not supposed to be solidifying on "arbitrary data", and if it is enabled then maybe it is because the operator cares about "arbitrary data"?

Bottom line, due to the `TipsSolidifier` activation config I am with @karimodm.

Reply:
good point @karimodm, still this will take time to fill (1 tx per scan). I still think that this can be avoided by separating the cases; it will reduce the amount of change.

However, I don't insist on this, if @GalRogozinski thinks it's fine then I'm OK.

Reply:
> still this will take time to fill (1 tx per scan)
Doesn't the `continue` here make the queue fill faster in any of the cases?

@GalRogozinski true on arbitrary data, this change combined with the requester freshness should fend off those scenarios though.